File: apps/ios/nearby/Services/Sui/RecipientService.swift
import Foundation
import LeanSuiApi

/// IO boundary for SuiNS forward resolution (`name.sui` → address). Wraps a GraphQL provider.
actor RecipientService {
  private let provider: GraphQLSuiProvider

  init(network: SuiNetworkKind) {
    self.provider = GraphQLSuiProvider(network: SuiNetwork(kind: network))
  }

  /// The address a SuiNS name points at, or nil if the name is unregistered. Uses the non-optional
  /// `…OrEmpty` variant so iOS and Android share one bridge-safe path ("" = unregistered).
  func resolve(name: String) async throws -> String? {
    let address = try await provider.resolveNameServiceAddressOrEmpty(name: name)
    return address.isEmpty ? nil : address
  }
}

File: apps/android/app/src/main/java/com/variance/nearby/services/sui/RecipientService.kt
package com.variance.nearby.services.sui

import com.variance.nearby.leansui.api.GraphQLSuiProvider
import com.variance.nearby.leansui.api.SuiNetwork
import kotlinx.coroutines.future.await
import org.swift.swiftkit.core.SwiftArena

/** IO boundary for SuiNS forward resolution (`name.sui` → address). Wraps a GraphQL provider. */
class RecipientService(
    network: SuiNetwork,
    swiftArena: SwiftArena,
) {
    private val provider: GraphQLSuiProvider = GraphQLSuiProvider.init(network, swiftArena)

    /** The address a SuiNS name points at, or null if the name is unregistered. Uses the non-optional
     *  `…OrEmpty` variant — the optional-`String` bridge mis-marshals on Android ("" = unregistered). */
    suspend fun resolve(name: String): String? = provider.resolveNameServiceAddressOrEmpty(name).await().takeIf { it.isNotEmpty() }
}
